macos/Sources/Ghostty/SurfaceView_AppKit.swift

```diff
@@ -572,6 +572,14 @@ extension Ghostty {
             bell = true
         }
 
+        @objc private func ghosttyBellDidReset() {
+            // Reset bell state
+            bell = false
+            // Clear dock badge
+            NSApp.dockTile.badgeLabel = nil
+            NSApp.dockTile.display()
```

Comment:
I don't think we should be modifying App state directly from the NSView here. It might be easier to just unconditionally reset `NSApp.dockTile` when the app becomes active in AppDelegate for now.